transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go

```diff
@@ -128,7 +129,18 @@ func getDotNetCoreVersion(mappingFile string, name string) (version string, err
 		logrus.Debugf("Could not load mapping at %s", mappingFile)
 		return "", err
 	}
-	return dotNetCoreNameVersionMapping.Spec.NameVersion[name], nil
+	var dotNetCoreFramework string
+	if dotnetcoreRegex1.MatchString(name) {
+		dotNetCoreFramework = dotnetcoreRegex1.FindString(name)
+	} else if dotnetcoreRegex2.MatchString(name) {
+		dotNetCoreFramework = dotnetcoreRegex2.FindString(name)
+	} else if dotnetcoreRegex3.MatchString(name) {
+		dotNetCoreFramework = dotnetcoreRegex3.FindString(name)
+	}
+	if _, ok := dotNetCoreNameVersionMapping.Spec.NameVersion[dotNetCoreFramework]; ok {
+		return dotNetCoreNameVersionMapping.Spec.NameVersion[dotNetCoreFramework], nil
+	}
+	return "", fmt.Errorf("could not find compatible .NET Core framework version")
```

Comment:
@Akash-Nayak the goal of the regex is to avoid the explicit version mappings requirements for all possible combination. We need to use some sort of heuristic to determine the best mapping. 

If the version numbers at the end of the name is always going to be the image name, and it is always supposed to be present, why not just get that value alone through regex and use that as the image name?

Reply:
Yes, I was also thinking that we should either have the version mappings or the regex-based heuristic. I have removed the version mappings now.